examples/static_map/count_by_key_example.cu

```diff
@@ -125,34 +126,37 @@ int main(void)
   // Compute capacity based on a 50% load factor
   auto constexpr load_factor = 0.5;
 
-  // If the number of unique keys is known in advance, we can use it to calculate the map capacity
-  std::size_t const capacity = std::ceil((num_keys / key_duplicates) / load_factor);
-  // If we can't give an estimated upper bound on the number of unique keys
-  // we conservatively assume each key in the input is distinct
-  // std::size_t const capacity = std::ceil(num_keys / load_factor);
+  // If the number of elements is known in advance, we can use it to calculate the map capacity
+  std::size_t const num_elements = num_keys / key_duplicates;
 
-  // Constructs a map with "capacity" slots.
-  cuco::static_map<Key, Count> map{
-    capacity, cuco::empty_key{empty_key_sentinel}, cuco::empty_value{empty_value_sentinel}};
+  // Constructs a map with number of elements and desired load factor.
+  auto map = cuco::experimental::static_map{
+    num_elements,
+    load_factor,
+    cuco::empty_key{empty_key_sentinel},
+    cuco::empty_value{empty_value_sentinel},
+    thrust::equal_to<Key>{},
```

Comment:
Side note 2: Spcifying `thrust::equal_to` feels weird. Should we add default values to the ref's tparams? Also, we should add an alias `cuco::default_key_equal = thrust::equal_to` of some sort.

Reply:
> Should we add default values to the ref's tparams?

IIRC, we cannot add default parameters to ref templates because of the variadic parameter in the end.

> we should add an alias cuco::default_key_equal = thrust::equal_to of some sort.

No preference here. OK to add but not required.